src/main/scala/com/yotpo/metorikku/output/catalog/TableMetadata.scala

```diff
@@ -19,21 +19,23 @@ class CatalogTable(tableName: String) {
     }
   }
 
-  def saveExternalTable(dataFrame: DataFrame, filePath: String, partitionBy: Option[Seq[String]], alwaysUpdateSchemaInCatalog: Boolean): Unit = {
+  def saveExternalTable(dataFrame: DataFrame, filePath: String, partitionBy: Option[Seq[String]],
+                        alwaysUpdateSchemaInCatalog: Boolean, saveMode: Option[String]): Unit = {
     val ss = dataFrame.sparkSession
     val catalog = ss.catalog
 
-    catalog.tableExists(tableName) match {
+    (catalog.tableExists(tableName), partitionBy, saveMode) match {
       // Quick overwrite (using alter table + refresh instead of drop + write + refresh)
-      case true => {
+      case (true, None, Some("Overwrite")) => {
```

Comment:
what if you have an overwrite partitioned output?

Reply:
I think saveMode can also be lower case "overwrite"

Reply:
Partitioned overwrite table should not be altered. That's the use case of segment events last